grype/pkg/syft_sbom_provider.go

```diff
@@ -106,7 +106,7 @@ func parseSBOM(scheme, path string) (io.Reader, *inputInfo, error) {
 	return r, info, nil
 }
 
-func decodeStdin(r io.Reader) (io.Reader, *inputInfo, error) {
+func decodeStdin(r io.Reader) (io.ReadSeeker, *inputInfo, error) {
 	b, err := io.ReadAll(r)
```

Comment:
Totally unrelated to this review, but it might be nice to be able to check whether the `ReadAll` is strictly necessary. This could already be a `ReadSeeker`, in which case `b` might be a large and unnecessary allocation.

Reply:
the reader being passed in is always stdin (an `*os.File`), which the types support the notion of seeking, however in practice if you try to seek at all you'll get an error. This seems like a bad idea from the stdlib to express stdin as a file at all (see https://github.com/golang/go/issues/13473).

Reply:
That makes sense - I hadn't noticed that it was always stdin. It also hadn't occurred to me that we might get a poisoned `ReadSeeker` (something that implements the interface but always errors if you call `Seek`).